File: hacking_with_python/cookie_theft.py
import requests
from colorama import init, Fore

# Initialize colorama for colored output
init(autoreset=True)

def test_cookie_theft_vulnerability(target_url):
    try:
        # Set headers to mimic a browser request
        headers = {
            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3'
        }

        # Send a GET request to the target website with custom headers
        response = requests.get(target_url, headers=headers, allow_redirects=True)

        # Check if the request was successful (status code 200)
        if response.status_code == 200:
            print(f"{Fore.GREEN}[+] Successfully accessed the website: {target_url}")
            
            # Check if the server returned any cookies
            if response.cookies:
                print(f"{Fore.GREEN}[+] Cookies received from the server:")
                for cookie in response.cookies:
                    print(f"\t{Fore.YELLOW}Name: {cookie.name}, Value: {cookie.value}")
            else:
                print(f"{Fore.RED}[-] No cookies received from the server.")
        else:
            print(f"{Fore.RED}[-] Failed to access the website: {target_url}. Status Code: {response.status_code}")

    except requests.exceptions.RequestException as e:
        print(f"{Fore.RED}[-] An error occurred: {str(e)}")


if __name__ == "__main__":
    print(f"{Fore.BLUE}[*] Welcome to Cookie Theft Vulnerability Tester")

    # Prompt user for the target website URL
    target_url = input(f"{Fore.BLUE}[*] Enter your target website URL: ")

    # Test the vulnerability
    test_cookie_theft_vulnerability(target_url)

File: hacking_with_ruby/cookie_theft.rb
require 'httparty'
require 'colorize'

def test_cookie_theft_vulnerability(url)
  begin
    response = HTTParty.get(url)
    if response.success?
      cookie = response.headers['set-cookie']
      if cookie.nil?
        puts "No cookies found in the response headers.".red
      else
        puts "Cookie Theft vulnerability test result for #{url}:".green
        puts "-----------------------------------------------".green
        puts "Cookie: #{cookie}".red
        puts "\n"
        puts "How to test this vulnerability:".blue
        puts "1. Copy the cookie value.".blue
        puts "2. Open another browser or incognito window.".blue
        puts "3. Paste the copied cookie value and access the website.".blue
        puts "4. If you can access the website without logging in, it indicates a Cookie Theft vulnerability.".blue
      end
    else
      puts "Failed to retrieve the page. HTTP Status Code: #{response.code}".red
    end
  rescue StandardError => e
    puts "An error occurred: #{e.message}".red
  end
end

puts "Enter the target website URL:".yellow
target_url = gets.chomp

test_cookie_theft_vulnerability(target_url)
